internal/provider/pipeline_resource_test.go

```diff
@@ -18,13 +20,13 @@ func TestPipelineResource(t *testing.T) {
 	})
 }
 func testPipeline() string {
-	return providerConfig + `
-data "warpstream_virtual_cluster" "default" {
-  default = true
-}
-
+	virtualClusterResource := fmt.Sprintf(`
```

Comment:
multiple tests are run in parallel, one for each terraform version: `1.0`, `1.1`, etc.

We need to make sure they aren't creating pipeline for the same default virtual cluster, which was causing errors. Here, we create a new virtual cluster instead.